tests/unit/components/NcAvatar/NcAvatar.spec.ts

```diff
@@ -20,11 +20,102 @@
  *
  */
 
-import { mount } from '@vue/test-utils'
+import { mount, shallowMount } from '@vue/test-utils'
 import { nextTick } from 'vue'
 import NcAvatar from '../../../../src/components/NcAvatar/NcAvatar.vue'
 
+interface ITestCase {
+	name: string
+	expected: string
+	props: Record<string, unknown>
+}
+
 describe('NcAvatar.vue', () => {
+	describe('Initials', () => {
+		const testCases: ITestCase[] = [
+			{
+				name: 'empty user',
+				expected: '?',
+				props: {},
+			},
+			{
+				name: 'user property',
+				expected: 'JD',
+				props: {
+					user: 'Jane Doe',
+					isNoUser: true,
+				},
+			},
+			{
+				name: 'display name property',
+				expected: 'JD',
+				props: {
+					displayName: 'Jane Doe',
+				},
+			},
+			{
+				name: 'display name property over user property',
+				expected: 'NU',
+				props: {
+					displayName: 'No User',
+					user: 'I am a group',
+					isNoUser: true,
+				},
+			},
+			{
+				name: 'special characters in name',
+				expected: 'JD',
+				props: {
+					displayName: 'Jane (Doe)',
+				},
+			},
+			{
+				name: 'lower case name',
+				expected: 'JD',
+				props: {
+					displayName: 'jane doe',
+				},
+			},
+			{
+				name: 'middle names',
+				expected: 'JD',
+				props: {
+					displayName: 'Jane Some Name Doe',
+				},
+			},
+			{
+				name: 'non ascii characters',
+				expected: 'ÜÖ',
+				props: {
+					displayName: 'Ümit Öçal',
+				},
+			},
+			{
+				name: 'non latin characters',
+				expected: 'ジド',
+				props: {
+					displayName: 'ジェーン ドー',
+				},
+			},
+		]
+
+		for (const { name, props, expected } of testCases) {
+			it(`can handle ${name}`, async () => {
+				const wrapper = shallowMount(NcAvatar, {
+					propsData: {
+						...props,
+					},
+				})
+				expect(wrapper.exists()).toBe(true)
+
+				await nextTick()
+				const initials = wrapper.find('.avatardiv__initials')
+				expect(initials.exists()).toBe(true)
+				expect(initials.text()).toBe(expected)
+			})
+		}
+	})
+
```

Comment:
These tests work fine, but manual generation of tests with a loop decrease DX a bit, for example, with testing GUI.

In WebStorm it doesn't work to go from test results to the test source, it cannot locate the generated test.

Both Jets and Vitest have `it.each` method to make tests by test cases, which seems more explicit to me.

With objects:

```suggestion
		it.each([
			{
				case: 'empty user',
				displayName: '',
				initials: '?',
			},
			{
				case: 'display name property',
				displayName: 'Jane Doe',
				initials: 'JD',
			},
			{
				case: 'special characters in name',
				displayName: 'Jane (Doe)',
				initials: 'JD',
			},
			{
				case: 'lower case name',
				displayName: 'jane doe',
				initials: 'JD',
			},
			{
				case: 'middle names',
				displayName: 'Jane Some Name Doe',
				initials: 'JD',
			},
			{
				case: 'non ascii characters',
				displayName: 'Ümit Öçal',
				initials: 'ÜÖ',
			},
			{
				case: 'non latin characters',
				displayName: 'ジェーン ドー',
				initials: 'ジド',
			},
		])('should display initials for $case ("$displayName" -> "$initials")', async ({ displayName, initials }) => {
			const wrapper = shallowMount(NcAvatar, {
				propsData: {
					displayName,
				},
			})
			await nextTick()
			expect(wrapper.text()).toBe(initials)
		})

		it('should display initials for user id', async () => {
			const wrapper = shallowMount(NcAvatar, {
				propsData: {
					user: 'Jane Doe',
					isNoUser: true,
				},
			})
			await nextTick()
			expect(wrapper.text()).toBe('JD')
		})

		it('should display initials for display name property over user id', async () => {
			const wrapper = shallowMount(NcAvatar, {
				propsData: {
					displayName: 'No User',
					user: 'I am a group',
					isNoUser: true,
				},
			})
			await nextTick()
			expect(wrapper.text()).toBe('NU')
		})
```

Reply:
Or with a template table (I'd prefer this option as more beautiful):

```suggestion
it.each`
			displayName             | initials | case
			${''}                   | ${'?'}   | ${'empty user'}
			${'Jane Doe'}           | ${'JD'}  | ${'display name property'}
			${'Jane (Doe)'}         | ${'JD'}  | ${'special characters in name'}
			${'jane doe'}           | ${'JD'}  | ${'lower case name'}
			${'Jane Some Name Doe'} | ${'JD'}  | ${'middle names'}
			${'Ümit Öçal'}          | ${'ÜÖ'}  | ${'non ascii characters'}
			${'ジェーン ドー'}        | ${'ジド'} | ${'non latin characters'}
		`('should display initials for $case ("$displayName" -> "$initials")', async ({ displayName, initials }) => {
			const wrapper = shallowMount(NcAvatar, {
				propsData: {
					displayName,
				},
			})
			await nextTick()
			expect(wrapper.text()).toBe(initials)
		})

		it('should display initials for user id', async () => {
			const wrapper = shallowMount(NcAvatar, {
				propsData: {
					user: 'Jane Doe',
					isNoUser: true,
				},
			})
			await nextTick()
			expect(wrapper.text()).toBe('JD')
		})

		it('should display initials for display name property over user id', async () => {
			const wrapper = shallowMount(NcAvatar, {
				propsData: {
					displayName: 'No User',
					user: 'I am a group',
					isNoUser: true,
				},
			})
			await nextTick()
			expect(wrapper.text()).toBe('NU')
		})
```

Reply:
In any case, I'd move those 2 tests from the loop over test cases, because they test different cases, not the initials generation algorithm.

Reply:
Links:
- https://jestjs.io/docs/api#testeachtablename-fn-timeout
- https://vitest.dev/api/#test-each

Reply:
Thank you! Did not know that something like this exists :see_no_evil: 
I also like the table style